omod/src/main/webapp/resources/scripts/components/LabTrackingOrderFactory.js

```diff
@@ -180,8 +184,13 @@ angular.module("labTrackingOrderFactory", [])
             order.locationWhereSpecimenCollected = order.location;
             order.proceduresForSpecimen = [];
             order.procedureNonCodedForSpecimen.value = order.procedureNonCoded.value;
-            order.postopDiagnosis.diagnosis.value = order.preLabDiagnosis.value;
-            order.postopDiagnosis.diagnosis.label = order.preLabDiagnosis.label;
+            if (typeof order.preLabDiagnosis === 'object') {
+              order.postopDiagnosis.diagnosis.value = order.preLabDiagnosis.value;
+              order.postopDiagnosis.diagnosis.label = order.preLabDiagnosis.label;
+            }
+            else {
+              order.postopDiagnosis.diagnosis =order.preLabDiagnosis;
+            }
```

Comment:
Also consider simply

```suggestion
            order.postopDiagnosis.diagnosis = order.preLabDiagnosis;
```

Reply:
I agree this is kind of ugly, unclear, but originally I had your suggested method and it wasn't working because it was copying via reference and I need to copy by value... but I should probably add a comment noting that to make it clearer.

Reply:
Oh yeah, that's reasonable. If it were known to be an object you could do `{ ...order.preLabDiagnosis }`, or if we were using lodash or Ramda you could use a clone function. But in the absence of those options this is good.

Reply:
Though you could just do `{ ...order.preLabDiagnosis }` within the is-object branch.

Reply:
Oh, right, thanks for the reminder @brandones

Reply:
Made that change, it makes it clearer the reason we are branching (added a comment as well)